src/Bicep.Core/Configuration/IConfigurationManager.cs

```diff
@@ -20,5 +19,43 @@ public interface IConfigurationManager
         /// <param name="sourceFileUri">The URI of the source file to get configuration for.</param>
         /// <returns>The configuration for the source file.</returns>
         RootConfiguration GetConfiguration(Uri sourceFileUri);
+
+        /// <summary>
+        /// Gets the built-in configuration.
+        /// </summary>
+        /// <returns>The built-in configuration.</returns>
+        static RootConfiguration GetBuiltInConfiguration() => BuiltInConfigurationLazy.Value;
+
+        public static IConfigurationManager ForConfiguration(RootConfiguration configuration) 
```

Comment:
What about `WithStaticConfiguration`? It's maybe a little unexpected that the `IConfigurationManager.GetConfiguration` method returned here doesn't respect the URI param.